services/viva-ms/src/lambdas/setCaseCompletions.ts

```diff
@@ -0,0 +1,77 @@
+import config from '../libs/config';
+
+import params from '../libs/params';
+import log from '../libs/logs';
+import { cases } from '../libs/query';
+
+import putVivaMsEvent from '../helpers/putVivaMsEvent';
+import completionsHelper from '../helpers/completions';
+import resetPersonSignature from '../helpers/resetPersonSignature';
+import { updateCaseCompletionStatus } from '../helpers/dynamoDb';
+
+import { CaseItem } from '../types/caseItem';
+import { VivaParametersResponse } from '../types/ssmParameters';
+
+type CaseKeys = Pick<CaseItem, 'PK' | 'SK'>;
+type UserCase = Pick<CaseItem, 'details' | 'persons' | 'currentFormId' | 'id'>;
+
+interface LambdaDetails {
+  caseKeys: CaseKeys;
+}
+
+export interface LambdaRequest {
+  detail: LambdaDetails;
+}
+
+export interface Dependencies {
+  getCase: (keys: CaseKeys) => Promise<UserCase>;
+  readParams: (envsKeyName: string) => Promise<VivaParametersResponse>;
+  putSuccessEvent: (params: LambdaDetails) => Promise<null>;
+  updateCase: (keys: CaseKeys, caseUpdateAttributes: unknown) => Promise<void>;
+}
+
+export const main = log.wrap(async event => {
+  return setCaseCompletions(event, {
+    getCase: cases.get,
+    readParams: params.read,
+    putSuccessEvent: putVivaMsEvent.setCaseCompletionsSuccess,
+    updateCase: updateCaseCompletionStatus,
+  });
+});
+
+export async function setCaseCompletions(event: LambdaRequest, dependencies: Dependencies) {
```

Comment:
In other implementations we can find `input` instead of `event`. event is AWS specific and maybe we like to not use AWS nameing?